client/ionic/src/content/flow.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  FLOWS_ENDPOINT,
+  FLOWS_PATH,
+  IMG_ENDPOINT,
+  IMG_PATH,
+} from './endpoints';
+import UserContext, { CONTEXT_DEFAULTS } from './userContext';
+import * as yaml from 'js-yaml';
+
+export interface BaseScreen {
+  type: string;
+  id: string;
+}
+
+// TODO: Should the texts be html instead?
+
+export interface TextImageScreen extends BaseScreen {
+  type: 'TextImage';
+  topImageUri?: string;
+  headingText: string;
+  bodyTexts?: string[];
+  bottomImageUri?: string;
+}
+
+// TODO: This is an example, and is not yet implemented.
+export interface HeroImageScreen extends BaseScreen {
+  type: 'HeroImage';
+  imageUri: string;
+}
+
+export type Screen = TextImageScreen | HeroImageScreen;
+
+export interface BaseFlow {
+  type: string;
+  screens: Screen[];
+}
+
+export interface CarouselFlow extends BaseFlow {
+  type: 'LinearCarousel';
+}
+
+export type Flow = CarouselFlow;
+
+export interface LoadedFlow {
+  readonly content: Flow;
+  readonly isLocal: boolean;
+  readonly imgPrefix: string;
+}
+
+export class FlowLoader {
+  // TODO: Disable pre-v0.1-ship if we are not confident in server support.
+  // TODO: A way to disable this easily in development.
+  static readonly ENABLE_SERVER_LOAD = true;
+
+  static readonly SUFFIX_DELIM = '.';
+
+  static contextSuffixes(ctx: UserContext): string[] {
+    // if you are in Canada using French, you would use the following content in
+    // priority order:
+    // ca.fr
+    // ca.en
+    // ww.fr
+    // ww.en
+    const suffixes = [
+      [ctx.country, ctx.language],
+      [ctx.country, CONTEXT_DEFAULTS.language],
+      [CONTEXT_DEFAULTS.country, ctx.language],
+      [CONTEXT_DEFAULTS.country, CONTEXT_DEFAULTS.language],
+    ];
+    return suffixes
+      .map(elems => elems.join(this.SUFFIX_DELIM))
+      .filter((val, idx, arr) => idx === arr.indexOf(val));
+  }
+
+  static async loadFlow(id: string, ctx: UserContext): Promise<LoadedFlow> {
+    const suffixes = this.contextSuffixes(ctx);
+    if (FlowLoader.ENABLE_SERVER_LOAD) {
+      // TODO: This is a lot of potential 404s. Consider loading a single
+      // manifest file (directory listing) that indicates which flows and
+      // suffixes exist.
+      for (const suffix of suffixes) {
+        try {
+          const url = `${FLOWS_ENDPOINT}/${id}.${suffix}.yaml`;
+          const respText = await (await fetch(url)).text();
```

Comment:
I don't think we need to nest await here.

Reply:
Yeah, I would have preferred something like this:
```suggestion
          const respText = await fetch(url).then(e => e.text());
```

Reply:
done